ztp/tools/pgt2acm/main.go

```diff
@@ -0,0 +1,421 @@
+	}
+	pathRelativeToOutputDir := filepath.Join(ACMTemplateDir, acmGenTempConversion.Policies[policyIndex].Manifests[manifestIndex].Path)
+	renamedpathRelativeToOutputDir, err := fileutils.RenderMCPLines(pathRelativeToOutputDir, policyGenTemp.Spec.Mcp)
+	if err != nil {
+		return fmt.Errorf("cannot render MCP lines, err: %s", err)
+	}
+	relativeManifestPath, err := filepath.Rel(filepath.Join(outputDir, relativePathTemplate), renamedpathRelativeToOutputDir)
+	if err != nil {
+		return fmt.Errorf("cannot get the relative path from path: %s and directory: %s, err: %s", renamedpathRelativeToOutputDir, outputDir, err)
+	}
+
+	// we switch to using the renamed manifest file with MCP line commented out
+	acmGenTempConversion.Policies[policyIndex].Manifests[manifestIndex].Path = relativeManifestPath
+	pathRelativeToOutputDir = renamedpathRelativeToOutputDir
+
+	// Unmarshal the manifest in order to check for metadata patch replacement
+	manifestFile, err := patches.UnmarshalManifestFile(pathRelativeToOutputDir)
+	if err != nil {
+		return fmt.Errorf("could not unmarshall manifest: %s, err: %s", pathRelativeToOutputDir, err)
+	}
+
+	if len(manifestFile) == 0 {
+		return fmt.Errorf("found empty YAML in the manifest at %s", pathRelativeToOutputDir)
+	}
+
+	kind, err := fileutils.GetManifestKind(pathRelativeToOutputDir)
+	if err != nil {
+		return fmt.Errorf("could not get manifest kind for file: %s, err: %s", pathRelativeToOutputDir, err)
+	}
+
+	if !stringhelper.StringInSlice[string](kindsToRender, kind.Kind, false) {
+		return nil
+	}
+
+	// Patch files only if needed
+	if len(acmGenTempConversion.Policies[policyIndex].Manifests[manifestIndex].Patches) == 0 || schema == "" {
+		return nil
+	}
+
+	patcher := patches.ManifestPatcher{Manifests: manifestFile, Patches: acmGenTempConversion.Policies[policyIndex].Manifests[manifestIndex].Patches}
+	const errTemplate = `failed to process the manifest at "%s": %w`
+
+	err = patcher.Validate()
+	if err != nil {
+		return fmt.Errorf(errTemplate, pathRelativeToOutputDir, err)
+	}
+
+	patchedFiles, err := patcher.ApplyPatches(schema)
+	if err != nil {
+		return fmt.Errorf(errTemplate, pathRelativeToOutputDir, err)
+	}
+	delete(patchedFiles[0], "apiVersion")
+	delete(patchedFiles[0], "kind")
+
+	acmGenTempConversion.Policies[policyIndex].Manifests[manifestIndex].Patches = patchedFiles
+	return nil
+}
+
+const (
+	waveAnnotationKey = "ran.openshift.io/ztp-deploy-wave"
+	sourceCrPrefix    = "source-crs"
+)
+
+// convertPGTPolicyToACMGenPolicy Converts PGT policy to ACM Gen policy
+//
+//nolint:funlen
+func convertPGTPolicyToACMGenPolicy(policyGenTemp *pgtformat.PolicyGenTemplate, baseDir, pgtPath, rootName, policyName, outputDir, forceWave string) (newPolicy acmformat.PolicyConfig, err error) {
+	newPolicy.Name = rootName + "-" + policyName
+	newPolicy.PolicyAnnotations = make(map[string]string)
+	wave := ""
+	for srcFileIndex := range policyGenTemp.Spec.SourceFiles {
```

Comment:
Can't this `for` be
```
for srcFileIndex, srcFile := range policyGenTemp.Spec.SourceFiles {
```
and then use `srcFile` throughout instead of always accessing it by using the index with `policyGenTemp.Spec.SourceFiles[srcFileIndex]`?

Reply:
This is a result from linting, the linter recommend using the index to avoid copying large  amount of data at each loop iteration (136 bytes)